### PR TITLE
Feature/window startup

### PR DESCRIPTION
**Pull request by FabioGNR**
User story #448 Startup
window creation and build fix

### CODE REVIEW

**Comment by notzain** [comment]
Check Travis logs: 
"default" is a reserved keyword, you are not allowed to use it as a parameter name.

File: engine/window/Window.h [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+class Window {
+public:
+    Window(WindowProperties& properties)
+        : m_properties{ properties }
+    {
+    }
+    virtual ~Window() = default;
+
+protected:
+    WindowProperties& m_properties;
```
Comment (notzain) [inline engine/window/Window.h]: Could be dangerous, dangling reference if the original WindowProperties goes out of scope. 

WindowProperties is trivially constructable, just copy it.

File: engine/window/SDLWindow.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+class SDLWindow : public Window {
+public:
+    SDLWindow(WindowProperties& properties);
+    ~SDLWindow() override{};
+
+private:
+    std::unique_ptr<SDL_Window, void (*)(SDL_Window*)> m_window;
+    int getFlags() const;
+    int getPosition(const int default) const;
```
Comment (notzain) [inline engine/window/SDLWindow.h]: default is a reserved keyword, use something else

Comment (FabioGNR) [reply engine/window/SDLWindow.h]: had wel aangepast in .cpp, maar .h vergeten. Pas ik aan

File: engine/window/Window.h (hunk 1 above)
Comment (FabioGNR) [reply engine/window/Window.h]: goed punt, thanks

**Comment by FabioGNR** [comment]
@notzain Ik heb de aanpassingen gemaakt en Travis build nu wel goed.
Kan je er nog is naar kijken?

**Comment by mattiejas** [comment]
Should this do anything? I don't see a window...

File: game/main.cpp [hunk 3]
```diff
@@ -1,9 +1,18 @@
...
 #include <iostream>
+#include <memory>
 
 int main()
 {
+    engine::WindowProperties windowProperties;
+    windowProperties.title = "Push'd OUT!";
+    windowProperties.maximized = true;
+    windowProperties.fullscreen = false;
+    std::unique_ptr<engine::Window> window{ new engine::SDLWindow(windowProperties) };
```
Comment (mattiejas) [inline game/main.cpp]:
```cpp 
new engine::SDLWindow(windowProperties)
```

Coupled too tight to SDL?

Maybe we could just call it `Window`?

**Comment by FabioGNR** [comment]
It works on my machine! ®

File: game/main.cpp (hunk 3 above)
Comment (FabioGNR) [reply game/main.cpp]: The abstract class is Window, but the specific implementation is SDLWindow. So it's not extremely coupled right?

Comment (mattiejas) [reply game/main.cpp]: Ah, my bad. Thought `SDLWindow` was from the library.

**Comment by notzain** [comment]
> Should this do anything? I don't see a window...

Builds for me, but same issue. Replacing WindowProperties with hard-coded values seems to fix the problem. 

Issue is that you never give windowProperties an width and height, so it is filled with random values. Filling out width/height fixes the problem

```cpp
    std::unique_ptr<engine::Window> window{
        new engine::SDLWindow({
            "Push'd OUT!",
            false, // full screen
            false, // maximized
            true, // centered
            640, // width
            480, // height
            0, // x
            0 // y
        })
    };
```

File: engine/window/WindowProperties.h [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+struct WindowProperties {
+    std::string title;
+    bool fullscreen;
+    bool maximized;
+    bool centered;
+    int width;
+    int height;
+    int x;
+    int y;
+};
```
Comment (notzain) [inline engine/window/WindowProperties.h]: Add default values.
```cpp
struct WindowProperties {
    std::string title = "";
    bool fullscreen = false;
    etc..
```

or value initialization
`int x{}`

File: game/main.cpp [hunk 5]
```diff
@@ -1,9 +1,18 @@
-#include <engine/physics/PhysicsManager.h>
+#include "engine/window/SDLWindow.h"
+#include "physics/PhysicsManager.h"
 #include <iostream>
+#include <memory>
 
 int main()
 {
+    engine::WindowProperties windowProperties;
```
Comment (notzain) [inline game/main.cpp]: `engine::WindowProperties windowProperties{};`

zero-initialization
https://blog.tartanllama.xyz/initialization-is-bonkers/

**Comment by FabioGNR** [comment]
I thought width and height were not required in this case because the flag MAXIMIZED is set. Though SDL does not say what is the behavior of the width&height in this case. I will use the zero-initialization for the properties object.

**Pull request merged**